apps/api/v1/views.py

```diff
@@ -50,6 +56,31 @@ def get_queryset(self):
             )
         return qs
 
+    def update(self, request, pk=None):
+        try:
+            obj = self.get_object()
+        except Http404:
+            # Using 406 as the 404 causes middleware issues for the pages.
```

Comment:
Could you explain this to me?

Reply:
pages.middleware.PageFallbackMiddleware fails to render `PageDetailView` unless we amend middleware to add another try and except it. It does that because it raises 404 and our middleware tries to find the right page for it.